docs/03-intermediate/03-wait-wake-system/wait_for.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 4
+hide_table_of_contents: true
+---
+
+# Breakdowns in communication 
+
+Sometimes the second program may not answer for any reason and it is necessary to be able to handle this case correctly. 
+
+## wait_for()
+
+Let's try to solve this problem using `wait_for(DURATION_IN_BLOCK)`.
+Reminder: this function itself wakes up the message from the waiting list after a certain number of blocks, if `exec::wake()` was not called.
+
+## Echo program
+
+Сomment out the reply line in the echo program so that the reply is not received: 
+
+```rust
+#[no_mangle]
+extern "C" fn handle() {
+    let message: String = msg::load().expect("Unable to decode");
+    // msg::reply(message, 0).expect("Error in sending a reply");
```

Comment:
this will still send an auto reply :( 
it needs long wait, for example, for your demonstration